test/unit/tap-reporter.js

```diff
@@ -0,0 +1,87 @@
+/* eslint-env mocha */
+
+var chai = require('chai')
+var sinon = require('sinon')
+var EventEmitter = require('events').EventEmitter
+var JsReporters = require('../../dist/js-reporters.js')
+var data = require('./data.js')
+var expect = chai.expect
+
+chai.use(require('sinon-chai'))
+
+describe('Tap reporter', function () {
+  var emitter
+
+  before(function () {
+    emitter = new EventEmitter()
+    var reporter = new JsReporters.TapReporter(emitter)
+  })
+
+  it('should output the TAP header', function () {
+    var spy = sinon.stub(console, 'log')
+
+    emitter.emit('runStart', {})
+    console.log.restore()
+
+    expect(spy).to.have.been.calledOnce
+  })
+
+  it('should output ok for a passing test', function () {
+    var spy = sinon.stub(console, 'log')
```

Comment:
It will not work with `beforeEach` and `afterEach`, because we will miss Mocha's reporter output for the test that has been exectued.

I think I found a way by wrapping the test function in `sinon.test`, by wrapping it up a sandbox is created on the function call and it is automatically restored on its end, this way we will not miss the reporter output. I will update the tests in 1 min.